src/dns-test.js

```diff
@@ -123,6 +135,50 @@ function encodeTCPQuery(domain, rrtype, dnssec_ok) {
     return buf;
 }
 
+/**
+ * Send a DNS query for an A record over UDP using the WebExtensions
+ * dns.resolve() API
+ *
+ * We query a random sub-domain under a domain name we control to ensure
+ * that our queries are not answered by the OS DNS cache. We do not seem
+ * to experience the same issue with the internal UDP/TCP APIs because
+ * they are not calling getaddrinfo().
+ *
+ * We let the underlying API handle re-transmissions and which nameserver is 
+ * used. We make sure that DoH is not used and that A records are queried, 
+ * rather than AAAA.
+ */
+async function sendUDPWebExtQuery(domain) {
+    let key = "udpAWebExt";
+    let errorKey = "AWebExt";
+    let flags = ["bypass_cache", "disable_ipv6", "disable_trr"];
+
+    let randomSubdomain;
+    let randomDomain;
+    try {
+        randomSubdomain = CRYPTO.randomBytes(16).toString('hex');
```

Comment:
That's a rather large dependency for just a random number. It's a oneliner in code, at a large code expense...

To generate a random number in a browser, use `crypto.getRandomValues` (and `.toString(16)` to convert each number to a hex string, followed by joining the array - note that toString won't pad zeroes but that doesn't really matter here). This doesn't take more than just a few lines of code at most.

If you really like to use a one-liner, then you could also reuse the `uuid` module that you're already using. Either by using the uuid as is (which has 128 bits of entropy, which is the 16 bytes that you're looking for), or by `.replaceAll,("-", "")` to only have the hex parts without the dashes.